From d3a0be54ac4625863a437b9f02dc87c75c3f5e7f Mon Sep 17 00:00:00 2001
From: David Wronek <david-w@ctemplar.com>
Date: Fri, 26 Nov 2021 16:09:44 +0100
Subject: [PATCH 200/210] arm64: dts: qcom: msm8916-huawei-y635: Add
 touchscreen

The Huawei Y635 has an MStar MSG2138 touchscreen, supplied by pm8916_l16
and pm8916_l6. Add it to the device tree.

Signed-off-by: David Wronek <david-w@ctemplar.com>
---
 .../boot/dts/qcom/msm8916-huawei-y635.dts     | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
index b52faa9a0d9..5c7a2f90cf1 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
@@ -72,6 +72,30 @@ proximity@39 {
 	};
 };
 
+&blsp_i2c5 {
+	status = "okay";
+
+	touchscreen@26 {
+		compatible = "mstar,msg2138";
+		reg = <0x26>;
+
+		interrupt-parent = <&msmgpio>;
+		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
+
+		reset-gpios = <&msmgpio 12 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&ts_int_reset_default>;
+
+		vdd-supply = <&pm8916_l16>;
+		vddio-supply = <&pm8916_l6>;
+
+		touchscreen-size-x = <2048>;
+		touchscreen-size-y = <2048>;
+
+		linux,keycodes = <KEY_BACK KEY_APPSELECT KEY_HOMEPAGE>;
+	};
+};
+
 &blsp1_uart2 {
 	status = "okay";
 };
@@ -248,4 +272,12 @@ prox_irq_default: prox-irq-default {
 		drive-strength = <2>;
 		bias-disable;
 	};
+
+	ts_int_reset_default: ts-int-reset-default {
+		pins = "gpio12", "gpio13";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
 };
-- 
2.34.1

